lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -223,24 +443,13 @@ def filter_query(
         and_list = []
         for kw in filters:
             combine_or = False
-            contains = False
-            try:
-                k, v = kw.split(":", maxsplit=1)
-            except ValueError:
-                raise APIAbort(
-                    HTTPStatus.BAD_REQUEST, f"filter {kw!r} must have the form 'k:v'"
-                )
-            if k.startswith("^"):
+            term = Term(kw).parse_filter()
+            if term.chain == "^":
                 combine_or = True
-                k = k[1:]
-            if v.startswith("~"):
-                contains = True
-                v = v[1:]
-
-            if not Metadata.is_key_path(k, Metadata.METADATA_KEYS, metalog_key_ok=True):
-                raise APIAbort(HTTPStatus.BAD_REQUEST, str(MetadataBadKey(k)))
-            keys = k.split(".")
+            keys = term.key.split(".")
             native_key = keys.pop(0).lower()
+            vtype = term.type if term.type else "str"
+            value = TYPES[vtype].convert(term.value, None)
```

Comment:
Maybe it's too much, but this strikes me as a nice thing to encapsulate inside the `Term` interface.  E.g.,
```python
            value = term.typed_value()
```
You could consider making it an functional attribute instead of a member function:  given that the `Term` instance already "knows" the `vtype` and the `value`, it doesn't need any arguments.

Reply:
Perhaps it's not worth it, given that the code at lines 466-469 uses `vtype` and `TYPES`...but it might be nice/fun anyway.

Reply:
This doesn't interest me, and as you said yourself "it's not worth it". The `Term` class helps with parsing the syntax, but while I considered also encapsulating much of the semantics on top of that I decided it wasn't worth going there at least for now.